pmxbot/core.py

```diff
@@ -430,9 +430,9 @@ def merge_dicts(a, b):
 
 
 class Bot(metaclass=abc.ABCMeta):
-	def out(self, channel, s, log=True):
+	def out(self, channel, s, log=True, context=None):
 		try:
-			sent = self.allow(channel, s) and self.transmit(channel, s)
+			sent = self.allow(channel, s) and self.transmit(channel, s, context)
```

Comment:
That wasn't caught in the functional tests because there were no functional tests that tested the bot responding to anything... that is until recently when in 2e2a2ef a regexp test was added. This test fails with the expected traceback in the bot:

```
Unhandled exception transmitting message: 'Clean up your language testingbot'
Traceback (most recent call last):
  File "/Users/jaraco/Dropbox/code/yg/public/pmxbot/pmxbot/core.py", line 438, in out
    sent = self.allow(channel, s) and self.transmit(channel, s, context)
TypeError: transmit() takes 3 positional arguments but 4 were given
```